### PR TITLE
fix(tasks): Handle non-string task env vars

### PR DESCRIPTION
**Pull request by icj217**
Project types that do not use `.projenrc.ts` cannot guard against end users providing numbers for environment variables. Casting them to strings before storing internally

If someone tries to do either of the following:
```js
// .projenrc.js
project.tasks.addEnvironment('LINT_THRESHOLD', 9);
```
```python
# .projenrc.py
project.tasks.add_environment('LINT_THRESHOLD', 9)
```

They'll get the following error on synthesis:
```
            if (value.startsWith("$(") && value.endsWith(")")) {
                      ^

TypeError: value.startsWith is not a function
    at RunTask.resolveEnvironment (/Users/.../node_modules/projen/lib/task-runtime.js:196:23)
    at new RunTask (/Users/.../node_modules/projen/lib/task-runtime.js:73:25)
    at TaskRuntime.runTask (/Users/.../node_modules/projen/lib/task-runtime.js:52:9)
    at Pip.installDependencies (/Users/.../node_modules/projen/lib/python/pip.js:51:17)
```

This PR makes the internal task components more defensive in terms of what kinds of values it might receive from end users. Added some tests to confirm behavior as well.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mrgrain** [comment]
I like this! Your implementation needs to be a bit more clever though in what values are accepted, see failing tests.

Also can we do an explicit check and emit a warning (via `this.project.logger.warn()`) if we encounter a conversion?

**Comment by icj217** [comment]
> I like this! Your implementation needs to be a bit more clever though in what values are accepted, see failing tests.
> 
> Also can we do an explicit check and emit a warning (via `this.project.logger.warn()`) if we encounter a conversion?

@mrgrain I've fixed the tests (had to exclude `undefined` from being stringified) and added the warning messages.

File: src/task.ts [hunk 1]
```diff
@@ -57,6 +58,7 @@ export class Task {
     this._locked = false;
 
     this._env = props.env ?? {};
+
```
Comment (mrgrain) [inline src/task.ts:61]: Should we do the same warning in the constructor?

File: src/task-runtime.ts [hunk 2]
```diff
@@ -255,7 +255,7 @@ class RunTask {
     const output: { [name: string]: string | undefined } = {};
 
     for (const [key, value] of Object.entries(env ?? {})) {
-      if (value.startsWith("$(") && value.endsWith(")")) {
+      if (String(value).startsWith("$(") && String(value).endsWith(")")) {
```
Comment (mrgrain) [inline src/task-runtime.ts:258]: Nice!

**Review by mrgrain — CHANGES_REQUESTED** [review]
Looking good. One last bit is missing.

File: src/task.ts (hunk 1 above)
Comment (icj217) [reply src/task.ts:62]: Good idea. I've added that private method

**Review by mrgrain — APPROVED** [review]

**Pull request merged**